api4/team_test.go

```diff
@@ -427,6 +427,74 @@ func TestPatchTeam(t *testing.T) {
 	CheckNoError(t, resp)
 }
 
+func TestRestoreTeam(t *testing.T) {
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+	Client := th.Client
+
+	createTeam := func(t *testing.T, deleted bool, teamType string) *model.Team {
+		t.Helper()
+		team := &model.Team{
+			DisplayName:     "Some Team",
+			Description:     "Some description",
+			CompanyName:     "Some company name",
+			AllowOpenInvite: (teamType == model.TEAM_OPEN),
+			InviteId:        model.NewId(),
+			Name:            "aa-" + model.NewRandomTeamName() + "zz",
+			Email:           "success+" + model.NewId() + "@simulator.amazonses.com",
+			Type:            teamType,
+		}
+		team, _ = Client.CreateTeam(team)
+		require.NotNil(t, team)
+		if deleted {
+			th.SystemAdminClient.SoftDeleteTeam(team.Id)
```

Comment:
Should assert success on the return values here.

Reply:
Thanks.  Check added https://github.com/mattermost/mattermost-server/pull/14297/commits/1937849dca14adb08e33634cbc30802f4fbbaa5d